hieroplatform/src/main/java/org/hiero/sketch/table/RowTableComparison.java

```diff
@@ -0,0 +1,72 @@
+package org.hiero.sketch.table;
+
+import org.hiero.sketch.spreadsheet.ColumnSortOrientation;
+import org.hiero.sketch.table.api.IColumn;
+import org.hiero.sketch.table.api.ITable;
+
+import java.util.Iterator;
+
+/**
+ * This class lets us compare a RowSnapShot to entries from a Table according to a prescribed
+ * RecordOrder. The schema of the snapshot should be a subset of that of the Table.
+ * This method is used to generate the NextK items in some order starting at a prescribed row by
+ * NextKSketch.
+ */
+public class RowTableComparison {
+    private final RowSnapshot topRow;
+    private final ITable table;
+    private final RecordOrder recordOrder;
+
+    public RowTableComparison(RowSnapshot topRow, ITable table, RecordOrder recordOrder ) {
+        this.topRow = topRow;
+        this.table = table;
+        this.recordOrder = recordOrder;
+    }
+
+    /**
+     * Compares Row i of Table to topRow according to recordOrder.
+     * @param i Index of a row in the Table.
+     * @return 1 if row i is greater, 0 if they are equal, -1 if it is less.
+     */
+    public int compareToRow(int i) {
```

Comment:
You have to settle on whether topRow can be null; in that case you should always return 1.